### PR TITLE
PHP 8.1 deprecations

### PR DESCRIPTION
**Pull request by claudiu-cristea**

### CODE REVIEW

**Comment by claudiu-cristea** [comment]
Some of the fixes, particularly `SiteAliasFileDiscovery` instantiations, might be fixed upstream in https://github.com/consolidation/site-alias/pull/46

**Comment by weitzman** [comment]
Thanks for the upstream PR. That looks like the best fix to me. I pinged @greg-1-anderson  there.

**Comment by greg-1-anderson** [comment]
Made releases for site-alias and site-process.

File: src/Preflight/Preflight.php [hunk 1]
```diff
@@ -237,7 +238,7 @@ public function preflight($argv): bool
         $paths = $this->configLocator->getSiteAliasPaths($this->preflightArgs->aliasPaths(), $this->environment);
 
         // Configure alias manager.
-        $aliasFileLoader = new \Drush\SiteAlias\SiteAliasFileLoader();
+        $aliasFileLoader = new \Drush\SiteAlias\SiteAliasFileLoader(new SiteAliasFileDiscovery([], '<= 1', ''));
```
Comment (greg-1-anderson) [inline src/Preflight/Preflight.php:241]: Should no longer be necessary.

File: src/Commands/core/SiteCommands.php [hunk 2]
```diff
@@ -166,7 +166,7 @@ public function siteAliasConvert($destination, $options = ['format' => 'yaml', '
         }
 
         // Configure legacy converter.
-        $discovery = new SiteAliasFileDiscovery();
+        $discovery = new SiteAliasFileDiscovery([], '<= 1', '');
```
Comment (greg-1-anderson) [inline src/Commands/core/SiteCommands.php:169]: Should no longer be necessary.

**Comment by greg-1-anderson** [comment]
Might want to pin to site-alias ^3.1.3 and remove constructor parameters here.

**Comment by weitzman** [comment]
Agreed. I just pushed an update for that.

**Pull request merged**